syntaxes/odoc.json

```diff
@@ -27,17 +27,17 @@
         },
         {
           "comment": "section header",
-          "begin": "\\{\\s\\{[[:digit:]]+\\}\\b",
-          "end": "\\}",
+          "begin": "[[:digit:]]\\b",
+          "end": "",
           "beginCaptures": {
             "1": { "name": "constant.language.odoc" }
           },
           "patterns": [{ "include": "#markup" }]
         },
         {
           "comment": "label",
-          "begin": "\\{\\s\\{[[:digit:]]+\\}\\s\\:\\s[label]\\b",
-          "end": "\\}",
+          "begin": "[[:digit:]]+\\b",
+          "end": "",
```

Comment:
You removed too much stuff. Only the `\\{` and `\\}` that were directly surrounding the `[[:digit:]]+` should have been removed (not all of them). And you also removed the label from the second part.